library/core/src/str/mod.rs

```diff
@@ -1524,7 +1524,8 @@ impl str {
     #[inline]
     pub fn split_once<'a, P: Pattern<'a>>(&'a self, delimiter: P) -> Option<(&'a str, &'a str)> {
         let (start, end) = delimiter.into_searcher(self).next_match()?;
-        Some((&self[..start], &self[end..]))
+        // SAFETY: `Searcher` is known to return valid indices.
+        unsafe { Some((self.get_unchecked(0..start), self.get_unchecked(end..self.len()))) }
```

Comment:
Why the change to passing ranges bounded on both ends, instead of just `..start` and `end..`?

Reply:
I looked through the `str/mod.rs` file and used this method with the same style: [1](https://github.com/rust-lang/rust/blob/0ba416580a547853189e82b4940b63afddda84ff/library/core/src/str/mod.rs#L573), [2](https://github.com/rust-lang/rust/blob/0ba416580a547853189e82b4940b63afddda84ff/library/core/src/str/mod.rs#L1983), [3](https://github.com/rust-lang/rust/blob/0ba416580a547853189e82b4940b63afddda84ff/library/core/src/str/mod.rs#L2090). So I'm not sure about this; possible solutions:
1. use ranges bounded on both ends everywhere;
2. use "minimal" ranges where possible on the whole file;
3. as you say, use not bounded ranges only in `split_once` and `rsplit_once`.